tests/core/test_document.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+
+from ragna.core import DocxDocumentHandler, LocalDocument
+
+
+@pytest.fixture()
+def docx_text():
+    return "Ragna is neat!"
+
+
+@pytest.fixture()
+def tmp_docx_document(tmp_path, docx_text):
+    import docx
+
+    document = docx.Document()
+    document.add_heading(docx_text)
+    document.add_paragraph(docx_text)
+    path = tmp_path / "test_document.docx"
+    document.save(path)
+    return LocalDocument.from_path(path)
+
+
+def test_docx(tmp_docx_document, docx_text):
+    handler = DocxDocumentHandler()
+    for page in handler.extract_pages(tmp_docx_document):
+        assert page.text == docx_text
```

Comment:
1. Instead of using the handler directly, you can also call `Document().extract_pages()`:

   https://github.com/Quansight/ragna/blob/c08a223e5eb6450681cabd7cfd5843c30dbe8d60/ragna/core/_document.py#L79-L80
2. We can harden this check by also checking that we get exactly one page back.

```suggestion
    pages = list(tmp_docx_document.extract_pages())
    assert len(pages) == 1
    
    page = pages[0]
    assert page.text == docx_text
```